### PR TITLE
serial-discovery now talks pluggable discovery protocol v1

### PR DESCRIPTION
**Pull request by cmaglie**
* Updated serial-discovery
* `discovery.Discovery` class now talks pluggable discovery [protocol v1](https://github.com/arduino/tooling-rfcs/pull/2)
* Also added a grpc `Port` for future use

This change should not break anything.

### CODE REVIEW

File: arduino/discovery/discovery.go [hunk 1]
```diff
@@ -117,24 +122,30 @@ func (disc *PluggableDiscovery) jsonDecodeLoop(in io.Reader, outChan chan<- *dis
 	closeAndReportError := func(err error) {
 		disc.statusMutex.Lock()
 		disc.alive = false
+		if debug {
+			fmt.Println("ERR:", err)
```
Comment (silvanocerza) [inline arduino/discovery/discovery.go:126]: Can't we just use `logrus.debug`?

File: arduino/cores/packagemanager/identify.go [hunk 2]
```diff
@@ -41,6 +41,7 @@ func (pm *PackageManager) IdentifyBoard(idProps *properties.Map) []*cores.Board
 		}
 		return false, true
 	}
+	// FIXME: TODO...
```
Comment (silvanocerza) [inline arduino/cores/packagemanager/identify.go:44]: I'd remove this, it scares me. :sweat_smile:

**Review by silvanocerza — CHANGES_REQUESTED** [review]
All in all it's fine for me, but I think debug printing could be done in a better way.

File: arduino/discovery/discovery.go (hunk 1 above)
Comment (cmaglie) [reply arduino/discovery/discovery.go:126]: I guess I'll just remove it...

File: arduino/cores/packagemanager/identify.go (hunk 2 above)
Comment (cmaglie) [reply arduino/cores/packagemanager/identify.go]: LOL, ok done

File: arduino/discovery/discovery.go (hunk 1 above)
Comment (silvanocerza) [reply arduino/discovery/discovery.go]: I very much prefer if we keep some debugging capabilities, I'd just prefer if we do it in a more structured way.

**Review by silvanocerza — APPROVED** [review]

**Pull request merged**